File: web-service/src/main/java/Post/AbstractPost.java
package Post;

import Files.UpFile;
import Notification.Notification;
import Notification.NotificationSubject;
import Registration.Group;
import User.User;
import core.SEntity;
import core.Searchable;
import core.hasPermission;

/**
 * a generic post abstract class to hold the common post data
 */
public abstract class AbstractPost extends SEntity implements NotificationSubject<User>, hasPermission, Searchable {

	User creator;
	String title, content;
	UpFile[] files;
	boolean isPublished;

	abstract protected Notification generateNotification(String title, String message);

	@Override
	public boolean add2NotificationList(User observer) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	@Override
	public boolean rmNotificationList(User observer) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	@Override
	public boolean notifyObservers(Notification Notification) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public Comment[] fetchComments() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public Group[] fetchGroups() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public UpFile[] fetchFiles() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public boolean addGroup(Group group) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public boolean addFile(UpFile file) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public boolean rmGroup(int id) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	public boolean rmFile(int id) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}
}

File: web-service/src/models/post/abstract-post.ts
import { User } from '../user/user';
import { SEntity } from '../core/s-entity';
import { NotificationSubject } from '../notification/notification-subject';
import { hasPermission } from '../core/has-permission';
import { Searchable } from '../core/searchable';
import { UpFile } from '../files/up-file';
import { Group } from '../registration/group';

/**
 * a generic post abstract class to hold the common post data
 */
export abstract class AbstractPost extends SEntity implements NotificationSubject<User>, hasPermission, Searchable<any> {
	public abstract hasPermission(): any;
	public abstract search(query: any): any;

	creator: User;
	title: string;
	content: string;
	files: UpFile[];
	isPublished: boolean;

	abstract generateNotification(title: string, message: string): Notification;

	public add2NotificationList(observer?: any): any {
		return null;
	}

	public rmNotificationList(observer?: any): any {
		return null;
	}

	public notifyObservers(Notification: Notification): boolean {
		return null;
	}

	public fetchComments(): Comment[] {
		return null;
	}

	public fetchGroups(): Group[] {
		return null;
	}

	public fetchFiles(): UpFile[] {
		return null;
	}

	public addGroup(group: Group): boolean {
		return null;
	}

	public addFile(file: UpFile): boolean {
		return null;
	}

	public rmGroup(id: number): boolean {
		return null;
	}

	public rmFile(id: number): boolean {
		return null;
	}

	constructor() {
		super();
		this.isPublished = false;
	}
}